src/cargo/ops/registry.rs

```diff
@@ -334,12 +335,13 @@ pub fn registry_configuration(
     Ok(RegistryConfig { index, token })
 }
 
-pub fn registry(
+fn registry(
     config: &Config,
     token: Option<String>,
     index: Option<String>,
     registry: Option<String>,
     force_update: bool,
+    validate_token: bool
```

Comment:
I added the token validation to `fn registry` because it seemed useful for several callers, but also because once the token is wrapped in a `Registry` private field we can't read it without bulkier changes that didn't seem justified.